### ISSUE TITLE
Web - Chat - Created Workspace in app does not show in Web immediately, only after refresh

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#11074]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Precondition: Log in with the same account in Web and app
1. Launch the app and open staging.new.expensify.com
2. Log in with the same expensifail account in app and Web
3. In app - tap on Fub menu
4. Select New Workspace
5. Check the Workspace under Setting in Web

## Expected Result:
Created Workspace in app is showing in Web immediately

## Actual Result:
Created Workspace in app does not show in Web immediately, only after refresh

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.2.1.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** applausetester+09162022abb@applause.expensifail.com/Feya86Katya

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/190812117-55acd453-e6fc-44cc-981a-05c168c8d743.MP4

https://user-images.githubusercontent.com/93399543/190812169-450033bb-5293-420a-864f-33cf9998a96e.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by PauloGasparSv** [comment]
Since we are refactoring the workspace creation right now, I'll link this issue there and let it be solved during the refactor! If this remains after the update we can re-open this and work on it separately!
https://github.com/Expensify/App/pull/10863

**Comment by neil-marcellini** [comment]
#10863 is for adding workspace **room**, not creating a workspace. A lot of people have made that mistake haha. Here is the tracking issue for creating workspaces https://github.com/Expensify/Expensify/issues/220859. Maybe @thienlnam will be fixing this as part of his follow up issues. If not, you should go for it.

**Comment by thienlnam** [comment]
Hmm, we use pusher to send back the new workspace details and so it should work https://github.com/Expensify/Web-Expensify/blob/abe2f6ed97456b84f4a9a9673af6ae0c71235b6f/lib/PolicyAPI.php#L241-L265

@kbecciv Are the default workspace chats created immediately as well or also have the same problem?

**Comment by thienlnam** [comment]
And yeah, like @neil-marcellini mentioned this is separate from the default workspace rooms and is already done being implemented so this should be investigated

**Comment by PauloGasparSv** [comment]
All right! Thks for the help @neil-marcellini @thienlnam :D, I'll investigate this!

**Comment by Puneet-here** [comment]
Hey I had reported it [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1663260126159989).

**Comment by PauloGasparSv** [comment]
@Puneet-here Idk if the speed the workspace chats load are directly linked to the workspace not showing in the user profile until we refresh the page! But I think I can investigate everything here.
If I find they are not linked I'll ping you so we can create another issue for that!

**Comment by thienlnam** [comment]
@Puneet-here The improvement you reported [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1663260126159989) isn't actually related to this current issue where the workspace created on mobile doesn't show up on Web.

We're also aware of the delay in workspace creation and have another issue pending that solves that UI issue https://github.com/Expensify/App/issues/9971 since the user will be immediately redirected to the policy from another page with this new design

**Comment by Puneet-here** [comment]
Ahhh sorry 😶‍🌫️ I saw the video and didn't read the steps carefully, you are right this is totally different.

**Comment by PauloGasparSv** [comment]
It seems like this is no longer happening on dev or staging anymore so I'm closing this issue here!

https://user-images.githubusercontent.com/6564265/192363511-a516dc2d-022e-4942-97b3-172f7b865aa9.mov

**Comment by thienlnam** [comment]
@PauloGasparSv From your video, I am not seeing you create the workspace from mobile - did you test that?

**Comment by PauloGasparSv** [comment]
Hey @thienlnam thanks for asking and sorry for taking so long!
I did not test on mobile, but now I tested the app against staging on my physical device and it works there too : ) 

https://user-images.githubusercontent.com/6564265/192803535-5f215593-dc3a-4ecb-b01e-569a3f9d15c2.MP4

**Comment by thienlnam** [comment]
Hey @PauloGasparSv, sorry for not clarifying in my comment but the issue is about device continuity not just the workspace showing up on the same device.
 
For example, if you create a workspace from your mobile app and are logged in on your computer on the Web, the new workspace should show immediately on Web as well.  If the created Workspace from your mobile app doesn't show up on Web without a refresh, that is a bug

**Comment by PauloGasparSv** [comment]
The new workspace/policy is being sent to all users successfully but  the [`CreateWorkspace` command](https://github.com/Expensify/Web-Expensify/blob/main/lib/PolicyAPI.php#L231) is not sending back the field `role` in the onyxData object. So the new workspace is missing because we [filter to only show workspaces/policies that we are admins of](https://github.com/Expensify/App/blob/main/src/pages/settings/InitialSettingsPage.js#L187) in the settings page.

Still thinking of how to fix this because simply sending the `makeMeAdmin` field to the `CreateWorkspace` command doesn't return the field back because of some other logic.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week) @PauloGasparSv ok to close this?

**Comment by PauloGasparSv** [comment]
@mvtglobally I think so, was waiting for it to be closed by the P.R.
Closing it here :D
